src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/impl/SwitchServiceImpl.java

```diff
@@ -518,6 +522,25 @@ public CompletableFuture<List<FlowPayload>> getFlowsForSwitch(SwitchId switchId,
                         .collect(Collectors.toList()));
     }
 
+    @Override
+    public CompletableFuture<SwitchFlowsPerPortResponse> getFlowsPerPortForSwitch(SwitchId switchId,
+                                                                                  Collection<Integer> ports) {
+        log.info("API request: Get all flows per port for the switch: {} with the port filter: {}", switchId, ports);
+        final String correlationId = RequestCorrelationId.getId();
+        GetFlowsPerPortForSwitchRequest data = new GetFlowsPerPortForSwitchRequest(switchId, ports);
+        CommandMessage message = new CommandMessage(data, System.currentTimeMillis(), correlationId);
+
+        return messagingChannel.sendAndGet(nbworkerTopic, message)
+                            .thenApply(GetFlowsPerPortForSwitchResponse.class::cast)
+                            .thenApply(GetFlowsPerPortForSwitchResponse::getFlowsByPorts)
+                            .thenApply(portToFlowDto -> portToFlowDto.entrySet().stream()
+                                    .collect(Collectors.toMap(entry -> entry.getKey(), entry -> entry.getValue()
+                                            .stream()
+                                            .map(flowMapper::toFlowResponseV2).collect(Collectors.toList()))))
+                            .thenApply(SwitchFlowsPerPortResponse::new);
```

Comment:
I think you can move mapping from `GetFlowsPerPortForSwitchResponse` to `SwitchFlowsPerPortResponse` into `switchMapping` module

Reply:
I moved the mapping into the `SwitchMapper`

Reply:
You didn't move it